shared/fetch_bsf.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2020 The WPT Dashboard Project. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+//go:generate mockgen -destination sharedtest/fetch_bsf_mock.go -package sharedtest github.com/web-platform-tests/wpt.fyi/shared FetchBSF
+
+package shared
+
+import (
+	"encoding/csv"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+const (
+	// experimentalBSFURL is the GitHub URL for fetching the experimental BSF data
+	// for Chrome, Firefox and Safari.
+	experimentalBSFURL = "https://raw.githubusercontent.com/Ecosystem-Infra/wpt-results-analysis/gh-pages/data/experimental-browser-specific-failures.csv"
+	// stableBSFURL is the GitHub URL for fetching the stable BSF data
+	// for Chrome, Firefox and Safari.
+	stableBSFURL = "https://raw.githubusercontent.com/Ecosystem-Infra/wpt-results-analysis/gh-pages/data/stable-browser-specific-failures.csv"
```

Comment:
Is there a rate limit for `raw.githubusercontent.com`? Shall we consider caching the responses?

Reply:
I think it falls under user-to-server requests for GitHub APIs, which should be 5000 calls/hour. But agreed, caching at the request level should be good for now